torchtext/vocab.py

```diff
@@ -139,33 +67,184 @@ def __init__(self, counter, max_size=None, min_freq=1, wv_dir=os.getcwd(),
             self.itos.append(k)
             self.stoi[k] = len(self.itos) - 1
 
-        if wv_type is not None:
-            self.set_vectors(wv_dict, wv_arr)
+        if vectors is not None:
+            self.load_vectors(vectors, unk_init=unk_init, expand_vocab=expand_vocab)
 
     def __len__(self):
         return len(self.itos)
 
-    def load_vectors(self, wv_dir=os.getcwd(), wv_type=None, wv_dim=300,
-                     unk_init='random'):
-        """Loads word vectors into the vocab
+    def load_vectors(self, vectors, unk_init='zero', expand_vocab=False):
+        """Arguments:
+              vectors: one of the available pretrained vectors or a list with each
+                  element one of the available pretrained vectors:
+                       glove.42B.300d
+                       glove.840B.300d
+                       glove.twitter.27B.25d
+                       glove.twitter.27B.50d
+                       glove.twitter.27B.100d
+                       glove.twitter.27B.200d
+                       glove.6B.50d
+                       glove.6B.100d
+                       glove.6B.200d
+                       glove.6B.300d
+                       charngram.100d
+              unk_init (string): 'zero' to initalize out-of-vocabulary word vectors
+                  to zero vectors; 'random' to initialize by drawing from a standard
+                  normal distribution
+              expand_vocab (bool): expand vocabulary to include all words for which
+                  the specified pretrained word vectors are available
+        """
+        if not isinstance(vectors, list):
+            vectors = [vectors]
+        vecs = []
+        tot_dim = 0
+        for v in vectors:
+            wv_type = v.split('.')[0]
+            wv_dim = int(v.split('.')[-1][:-1])
+            if wv_type == 'glove':
+                wv_name = '.'.join(v.split('.')[1:-1])
+                vecs.append(GloVe(name=wv_name, dim=wv_dim, unk_init=unk_init))
+                if expand_vocab:
+                    for w in sorted(vecs[-1].stoi.keys()):
+                        self.itos.append(w)
+                        self.stoi[w] = len(self.itos) - 1
+            elif 'charngram' in v:
+                vecs.append(CharNGram(unk_init=unk_init))
+            tot_dim += wv_dim
+
+        self.vectors = torch.Tensor(len(self), tot_dim)
+        start_dim = 0
+        for i, token in enumerate(self.itos):
+            for i, v in enumerate(vectors):
+                end_dim = start_dim + vecs[i].dim
+                self.vectors[i][start_dim:end_dim] = vecs[i][token]
+                start_dim = end_dim
+            assert(start_dim == tot_dim)
+            start_dim = 0
+
+    def set_vectors(self, stoi, vectors, dim, unk_init='zero'):
+        self.vectors = torch.Tensor(len(self), dim)
+        self.vectors.normal_(0, 1) if unk_init == 'random' else self.vectors.zero_()
```

Comment:
I'm not actually using this; just left it in case someone wanted to manually set the vectors. Not sure what you mean by parameterizing it though. Since I'm not using it, I'm very open to changing it around. Just let me know what you're thinking.